_posts/2024-05-04-openloco-v24.05.md

```diff
@@ -0,0 +1,149 @@
+
+Side note: a basic window takes up about 2kB of RAM, not including widgets. This means vanilla was using 24kB of
+memory for windows. Thankfully, memory isn't that scarce any more, so we're affording a whopping 128kB now.
+
+## Correct naming and use of ViewportVisibility::heightMarksOnTrack (#2468)
+
+Remember how we [changed the granualarity](https://openloco.io/news/2024/01/openloco-v24.01.html#make-viewport-see-through-flags-more-granular-2231)
+of the see-through viewport flags earlier this year? Well, it seems we missed a bit,
+which meant the surface height markers were appearing during construction, when track height markers
+were meant to be shown!
+
+Interestingly, this is down to a much older mistake in our reimplementation, where we had swapped the
+two flags internally. Only now that our implementation is furthering do these mistaken labels start
+having an effect!
+
+## Implement tile element insertion (#2415, #2421)
+
+In grid-based games like Locomotion, tile elements comprise everything that is or can be placed on the map.
+Oddly enough, though, we hadn't gotten around to reimplementing the function that actually allocates these
+elements. Instead, we just continued to use the vanilla implementation. That is no longer needed now.
+
+Last time, we mentioned we were working on [No Locomotion](https://openloco.io/news/2024/04/openloco-v24.04.html#work-on-no-locomotion-executables-2395) versions of the game.
+These versions do not call into vanilla Locomotion functions at all, and highlight areas that our code doesn't
+cover yet. Tile element insertion was one of these areas. This means we can now actually get in-game!
+
+## Implement number formatting (#2118, #2453)
+
+Whenever strings are displayed in-game, they go through a _formatting_ process. This process can change
+text colours or fonts, but also fill in placeholders with actual data. This can be the name of an industry,
+but also a numerical value, including currency. Turns out we hadn't actually reimplemented number formatting
+in C++!
+
+Formatting is a surprisingly controversial topic, what with different cultures using different ways of
+separating decimals, or indeed grouping thousands -- or indeed not grouping by thousands, but ten-thousands
+(Chinese, Japanese, Korean).
+
+We've elected to defer number formatting to a library we had previously incorporated, libfmt.
+While we currently only support grouping numbers by thousands, we aim to support alternatives in the future,
+which should be made easier through this library.
+
+## Rework string formatting (#2461, #2464, #2469)
+
+With the last formatting codes reimplemented, we could finally refactor the way strings are reformatted.
+The previous approach was very similar to vanilla, which wasn't actually that memory safe. This mostly
+showed in to the buffer length not being taken into account at all times. This could be problematic in
```

Comment:
This sounds odd

Reply:
Yeah, that 'to' isn't supposed to be there. Thanks.